src/app/settings/action-triggers/components/edit-action-triggers/edit-action-triggers.component.ts

```diff
@@ -0,0 +1,174 @@
+import { AbstractControl, FormBuilder, UntypedFormArray, UntypedFormControl, UntypedFormGroup, Validators } from '@angular/forms'
+import { ActionTrigger, ActionTriggerWithActionInfo, CreateActionTrigger, KeyboardAndSelectionTriggerData, KeyboardTriggerData, SHORTCUT_KEYS_MAPPINGS } from 'src/app/shared/models/action-trigger'
+import { Tv2PartAction } from 'src/app/shared/models/tv2-action'
+import { ActionTriggerService } from 'src/app/shared/abstractions/action-trigger.service'
+
+export interface ActionTriggerUIForm {
+  actionId: string
+  data: {
+    label: string
+    keys: string[]
+    actionArguments: number
+  }
+}
+
+@Component({
+  selector: 'sofie-edit-action-triggers',
+  templateUrl: './edit-action-triggers.component.html',
+  styleUrls: ['./edit-action-triggers.component.scss'],
+})
+export class EditActionTriggersComponent implements OnChanges {
+  @Output() private readonly cancelActionTrigger: EventEmitter<void> = new EventEmitter<void>()
+  @Input() public selectedActionTrigger: ActionTriggerWithActionInfo<KeyboardAndSelectionTriggerData> | null
+  @Input() public actions: Tv2PartAction[]
+  public submitting: boolean = false
+  public submitted: boolean = false
+  public keyPress: boolean = false
+  public selectedAction: Tv2PartAction | null
+
+  constructor(
+    private readonly fb: FormBuilder,
+    private readonly actionTriggerService: ActionTriggerService
+  ) {}
+
+  public actionForm: UntypedFormGroup = this.fb.group({
+    actionId: ['', [Validators.required]],
+    data: this.fb.group({
+      keys: this.fb.array([]),
+      actionArguments: '',
+      label: [''],
+    }),
+  })
+
+  public ngOnChanges(changes: SimpleChanges): void {
+    const actionTriggerChange: SimpleChange | undefined = changes['selectedActionTrigger']
+    if (actionTriggerChange) {
+      const action: ActionTriggerWithActionInfo<KeyboardAndSelectionTriggerData> | null = actionTriggerChange.currentValue
+      this.clearFormArray(this.attributesArray)
+      if (action) {
+        this.patchValue(action)
+        this.selectedAction = action.actionInfo
+      } else {
+        this.actionForm.reset()
+        this.selectedAction = null
+      }
+    }
+  }
+
+  private patchValue(actionValue: ActionTrigger<KeyboardAndSelectionTriggerData>): void {
+    this.actionForm.patchValue(actionValue)
+    this.addArrValues(actionValue.data.keys, this.attributesArray)
+  }
+
+  private clearFormArray(formArray: UntypedFormArray): void {
+    while (formArray.length !== 0) {
+      formArray.removeAt(0)
+    }
+  }
+
+  public onKeyDown(event: KeyboardEvent): void {
+    event.preventDefault()
+    const newKeyCode: string = SHORTCUT_KEYS_MAPPINGS[event.code] ? SHORTCUT_KEYS_MAPPINGS[event.code] : event.code
+    if (this.keyPress) {
+      const currentKeys: string[] = this.actionForm?.get('data')?.get('keys')?.value ? this.actionForm?.get('data')?.get('keys')?.value : []
+      if (currentKeys.findIndex(keyCode => keyCode === newKeyCode) === -1) {
```

Comment:
To make the code a bit more saying in a human readable way, when we check for the existence of items in an array, we can use `Array.some` or `Array.every`. This would become:
```suggestion
      if (currentKeys.every(keyCode => keyCode !== newKeyCode)) {
```

Reply:
Done